src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -283,27 +296,43 @@ def register_blob_source_uri_arguments(scope):
-
-for item in ['create', 'update']:
-    register_cli_argument('storage account {}'.format(item), 'sku', help='The storage account SKU.', **model_choice_list(ResourceType.MGMT_STORAGE, 'SkuName'))
-    es_model = get_sdk(ResourceType.MGMT_STORAGE, 'models#EncryptionServices')
-    if es_model:
-        register_cli_argument('storage account {}'.format(item), 'encryption', nargs='+', help='Specifies which service(s) to encrypt.', validator=validate_encryption, **enum_choice_list(list(es_model._attribute_map.keys())))  # pylint: disable=protected-access
-
-register_cli_argument('storage account create', 'access_tier', help='Required for StandardBlob accounts. The access tier used for billing. Cannot be set for StandardLRS, StandardGRS, StandardRAGRS, or PremiumLRS account types.', **model_choice_list(ResourceType.MGMT_STORAGE, 'AccessTier'))
-register_cli_argument('storage account update', 'access_tier', help='The access tier used for billing StandardBlob accounts. Cannot be set for StandardLRS, StandardGRS, StandardRAGRS, or PremiumLRS account types.', **model_choice_list(ResourceType.MGMT_STORAGE, 'AccessTier'))
-register_cli_argument('storage account create', 'custom_domain', help='User domain assigned to the storage account. Name is the CNAME source.')
-register_cli_argument('storage account update', 'custom_domain', help='User domain assigned to the storage account. Name is the CNAME source. Use "" to clear existing value.', validator=validate_custom_domain)
-register_cli_argument('storage account update', 'use_subdomain', help='Specify whether to use indirect CNAME validation.', **enum_choice_list(['true', 'false']))
-
-register_cli_argument('storage account update', 'tags', tags_type, default=None)
-register_cli_argument('storage account update', 'https_only', help='Allows https traffic only to storage service.', **three_state_flag())
+
+def register_common_storage_account_options(c):
+    c.reg_arg('https_only', help='Allows https traffic only to storage service.', **three_state_flag())
+    c.reg_arg('sku', help='The storage account SKU.', **model_choice_list(ResourceType.MGMT_STORAGE, 'SkuName'))
+    c.reg_arg('encryption', nargs='+', help='Specifies which service(s) to encrypt.', validator=validate_encryption,
+              resource_type=ResourceType.MGMT_STORAGE, min_api='2016-12-01',
+              **enum_choice_list(list(get_sdk(ResourceType.MGMT_STORAGE,
+                                              'models#EncryptionServices')._attribute_map.keys())))  # pylint: disable=protected-access
+    c.reg_arg('assign_identity', help='Generate and assign a new Storage Account Identity for this storage account for '
+                                      'use with key management services like Azure KeyVault.',
+              action='store_true', resource_type=ResourceType.MGMT_STORAGE, min_api='2017-06-01')
```

Comment:
Since you don't use IGNORE_TYPE, wouldn't --assign-identity still appear in older api versions since it's in the command handler signature?